README.md

```diff
@@ -6,22 +6,78 @@ Copyright (c) 2022-2023 [Check Digit, LLC](https://checkdigit.com)
+- `--entryPoint` the entry point for the bundle, relative to the inDir. if not provided, the files in the inDir will
+  be processed as individual unbundled files.
+- `--inDir` the input source code directory.
+- `--outDir` the output directory.
+- `--outFile` the output file, relative to `--outDir`. This is provided for single-file bundles, along with `--entryPoint`.
+- `--external` external modules to exclude from the bundle. Built-in `node` modules are automatically excluded.
+  A wildcard `*` can be used to exclude multiple external modules.
+- `--minify` whether to minify the output.
+- `--sourceMap` whether to include inline sourcemap.
+
+#### Examples
+
+```
+# build commonjs .cjs files from Typescript source
+npx builder --type=commonjs --outDir=build-cjs
+
+# build single-file commonjs .cjs bundle from Typescript source
+npx builder --type=commonjs --entryPoint=index.ts --outDir=build-cjs-bundle --outFile=index.cjs
+
+# build ESM .mjs files from Typescript source
+npx builder --type=module --outDir=build-esm
+
+# build single-file ESM .mjs bundle from Typescript source
+npx builder --type=module --outDir=build-esm-bundle --entryPoint=index.ts --outFile=index.mjs
+```
+
+### Tests
+
+This module includes a number of integration-style tests, to ensure that a specific version of Typescript will interoperate
+with `builder`, in addition to libraries and frameworks used by Check Digit:
+
+- Jest and `ts-jest`
+- ESLint and `@typescript-eslint/eslint-plugin`
+- Built-in `node:test` runner
+- prettier
+- tsc
+- esbuild
+
+We do this to ensure that Typescript upgrades do not break these dependencies. New major versions of Typescript are not immediately
+supported by projects such as ts-jest, eslint, prettier, etc. Our policy is to wait until these projects fully support
+the new version of Typescript, and/or without emitting warnings during these tests, before publishing.
 
-#### A note about versioning
+### A note about versioning
 
 Strict semver is a little complicated, as Typescript itself does not adhere to semver. So our "best effort" policy is:
 
-- Each new target (e.g. `ES2019` to `ES2020`) will result in a new major version of this module. We coordinate this
-  with whatever the latest LTS version of Node is currently supported by Amazon Lambda, Google Cloud Functions
+- Each update to the minimum Node target (e.g. Node 16 to Node 18) will result in a new major version of this module.
+  We coordinate this with whatever the latest LTS version of Node is currently supported by Amazon Lambda, Google Cloud Functions
   and Azure Functions.
-- Each new major version of Typescript (e.g. `4.2.x` to `4.3.x`) will result in a new minor version of this module.
-- Each new minor update of Typescript (e.g. `4.3.1` to `4.3.2`) will result in a new patch version of this module.
+- Each new "major" version of Typescript (e.g. `4.2.x` to `4.3.x`) will result in a new minor version of this module.
```

Comment:
the examples doesn't seem match with the description, should it be "(e.g. `3.x.x` to `4.x.x`)" instead?

Reply:
4.2 to 4.3 is a major version update to typescript.  it doesn't follow semver.